docs/manual/java/guide/cluster/PersistentEntityCassandra.md

```diff
@@ -22,4 +22,46 @@ In sbt:
 
 ## Configuration
 
+Lagom uses several internal Cassandra tables to persist entity data. These are known as the journal, which stores events, and the snapshot store, which stores snapshots of the state as an optimization for faster recovery (see [[Snapshots|PersistentEntity#Snapshots]] for details). [[Cassandra Read-Side support|ReadSideCassandra]] also uses a table to store the event offsets last processed by each read-side processor (detailed in [[Read-side design|ReadSide#Read-side-design]]). You will need to configure the keyspaces that contain the tables for each of these components.
+
+A keyspace in Cassandra is a namespace that defines data replication on nodes. Each service should use a unique keyspace name so that the tables of different services do not conflict with each other.
+
+You can configure these keyspaces in each service implementation project's `application.conf` file:
+
+```conf
+cassandra-journal.keyspace = users_journal
+cassandra-snapshot-store.keyspace = users_snapshot
+lagom.persistence.read-side.cassandra.keyspace = users_read_side
+```
+
+Cassandra keyspace names must start with an alphanumeric character and contain only alphanumeric and underscore characters. They are case-insensitive and stored in lowercase.
+
+While different services should be isolated by using different keyspaces, it is perfectly fine to use the same keyspace for all of these components within one service. In that case, it can be convenient to define a custom keyspace configuration property and use [property substitution](https://github.com/typesafehub/config#factor-out-common-values) to avoid repeating it.
+
+```conf
+users.cassandra.keyspace = users
+
+cassandra-journal.keyspace = ${users.cassandra.keyspace}
+cassandra-snapshot-store.keyspace = ${users.cassandra.keyspace}
+lagom.persistence.read-side.cassandra.keyspace = ${users.cassandra.keyspace}
+```
+
+By default, Lagom automatically creates the configured keyspaces and its internal tables if they are missing. If you prefer to manage the schema explicitly, you can disable automatic creation with these properties:
```

Comment:
There's a distinction between specifying the name of the keyspace and whether Lagom automatically creates it or not.

Either way, you need to configure the keyspace names.

By default, Lagom will look for keyspaces with those names and create them if they're missing. If you set the `keyspace-autocreate` to `false`, then it will just assume that the keyspaces you specify are present already, and fail with an error if they're not. (To be pedantic, it will fail and retry a certain number of times, so you can technically create the keyspace concurrently with the service starting.)

So unless I'm misunderstanding what you mean, I think this order makes more sense: the `autocreate` properties depend on and extend the behavior of the `keyspace` properties. The `autocreate` properties are optional to configure, while the `keyspace` ones are (becoming) mandatory.

(I just noticed that my example got everything backwards and should have set them to `false` rather than `true`.)